dids/src/main/kotlin/web5/sdk/dids/DidKey.kt

```diff
@@ -0,0 +1,130 @@
+package web5.sdk.dids
+
+import com.nimbusds.jose.Algorithm
+import com.nimbusds.jose.JWSAlgorithm
+import com.nimbusds.jose.jwk.Curve
+import foundation.identity.did.DID
+import foundation.identity.did.DIDDocument
+import foundation.identity.did.VerificationMethod
+import io.ipfs.multibase.Multibase
+import web5.sdk.common.Varint
+import web5.sdk.crypto.Crypto
+import web5.sdk.crypto.KeyManager
+import java.net.URI
+
+// TODO: move this to Crypto
+private val CURVE_CODEC_IDS = mapOf(
+  Curve.Ed25519 to Varint.encode(0xed),
+  Curve.SECP256K1 to Varint.encode(0xe7)
+)
+
+/**
+ * Represents options for creating "did:key" DIDs with specified cryptographic configurations.
+ *
+ * This class is designed to hold specific configurations, namely the [algorithm] and [curve],
+ * during the creation of a new "did:key" DID using the [DidKeyMethod] implementation.
+ *
+ * @property algorithm A JOSE [Algorithm] to be used for generating the key (defaults to EdDSA).
+ * @property curve A cryptographic [Curve] to be used for key generation (defaults to Ed25519).
+ *
+ * ### Example Usage:
+ * ```
+ * val options = CreateDidKeyOptions(algorithm = JWSAlgorithm.EdDSA, curve = Curve.Ed25519)
+ * ```
+ */
+public class CreateDidKeyOptions(
+  public val algorithm: Algorithm = JWSAlgorithm.EdDSA,
+  public val curve: Curve = Curve.Ed25519,
+) : CreateDidOptions
+
+public typealias DidKey = Did
+
+/**
+ * Provides a specific implementation for creating and resolving "did:key" method Decentralized Identifiers (DIDs).
+ *
+ * The "did:key" method is a specific DID method that is intended for use with
+ * DIDs that are entirely derived from a single public key, and it's described in detail
+ * in the W3C DID specification.
+ *
+ * ### Usage Example:
+ * ```
+ * val keyManager = InMemoryKeyManager()
+ * val did = DidKeyMethod.create(keyManager)
+ * ```
+ */
+public object DidKeyMethod : DidMethod<CreateDidKeyOptions> {
+  override val method: String = "key"
+
+  /**
+   * Creates a new "did:key" DID, derived from a public key, and stores the associated key in the provided [KeyManager].
+   *
+   * The method-specific identifier of a "did:key" DID is a multibase encoded public key.
+   *
+   * **Note**: Defaults to Ed25519 if no options are provided
+   *
+   * @param keyManager A [KeyManager] instance where the new key will be stored.
+   * @param options Optional parameters ([CreateDidKeyOptions]) to specify algorithm and curve during key creation.
+   * @return A [DidKey] instance representing the newly created "did:key" DID.
+   *
+   * @throws UnsupportedOperationException if the specified curve is not supported.
+   */
+  override fun create(keyManager: KeyManager, options: CreateDidKeyOptions?): DidKey {
```

Comment:
Having looked at a couple of standard libs (like `HttpClient` from ktor and `Json` from `kotlinx.serialization`), I've noticed that a common pattern is to use a builder configuration object.

That's the approach I took in #50 . Specifically, the configuration object is defined as in https://github.com/TBD54566975/web5-sdk-kotlin/blob/2bb4d2a54e9e9b032e296ba084daf04c9c74fcdf/dids/src/main/kotlin/DIDIonManager.kt#L62-L68

it's applied as shown in https://github.com/TBD54566975/web5-sdk-kotlin/blob/2bb4d2a54e9e9b032e296ba084daf04c9c74fcdf/dids/src/main/kotlin/DIDIonManager.kt#L74-L77

Additionally, there is a default, defined in https://github.com/TBD54566975/web5-sdk-kotlin/blob/2bb4d2a54e9e9b032e296ba084daf04c9c74fcdf/dids/src/main/kotlin/DIDIonManager.kt#L263

Doing that, allows you to use as show in the following tests. 

1. What it looks like in tests when using the default object https://github.com/TBD54566975/web5-sdk-kotlin/blob/2bb4d2a54e9e9b032e296ba084daf04c9c74fcdf/dids/src/test/kotlin/DIDIonTest.kt#L30

2. And then when defining some options within it https://github.com/TBD54566975/web5-sdk-kotlin/blob/2bb4d2a54e9e9b032e296ba084daf04c9c74fcdf/dids/src/test/kotlin/DIDIonTest.kt#L41-L52

It seems more idiomatic kotlin, but I have limited experience. What are your thoughts?

Reply:
that said, there's nothing precluding a concrete DID implementation from providing builder functionality in addition to `create`. i just don't want to enforce it at the interface level